pkg/chartsync/git.go

```diff
@@ -281,12 +291,22 @@ func (c *GitChartSync) sync(hr *v1.HelmRelease, mirrorName string, repo *git.Rep
 // maybeMirror requests a new mirror for the given remote. The return value
 // indicates whether the repo was already present (`true` if so,
 // `false` otherwise).
-func (c *GitChartSync) maybeMirror(mirrorName string, remote string) bool {
+func (c *GitChartSync) maybeMirror(mirrorName string, source *v1.GitChartSource, namespace string) bool {
+	gitURL := source.GitURL
+	if source.SecretRef != nil {
```

Comment:
As you are doing the same `nil` check in `addAuthForHTTPS`, I would reduce this to

```go
gitURL := source.GitURL
var err error
if gitURL, err = c.addAuthForHTTPS(source.GitURL, source.SecretRef, namespace); err != nil {
  c.logger.Log("error", GitAuthError{err}.Error())
  return false
}
```

Reply:
A good suggestion, changed.